include/mc_rtc/Configuration.h

```diff
@@ -102,24 +102,59 @@ struct has_configuration_save_object : decltype(_has_configuration_save_object::
  */
 struct MC_RTC_UTILS_DLLAPI Configuration
 {
+private:
+  /*! \brief Implementation details
+   *
+   * This structure is meant to hide the JSON library used by mc_rtc
+   * while allowing the template method implementation.
+   */
+  struct MC_RTC_UTILS_DLLAPI Json
+  {
+    bool isArray() const;
+    size_t size() const;
+    Json operator[](size_t idx) const;
+    bool isObject() const;
+    std::vector<std::string> keys() const;
+    Json operator[](const std::string & key) const;
+    bool isNumeric() const;
+    double asDouble() const;
+    void path(std::string & out) const;
+    void * value_;
+    std::shared_ptr<void> doc_;
+  };
 
+public:
   /*! \brief Exception thrown by this class when something bad occurs
    */
   struct MC_RTC_UTILS_DLLAPI Exception : public std::exception
```

Comment:
A small comment about the cost of generating the exception message when it is not silenced might be good. Even though in most cases it won't matter.